### PR TITLE
Added Linear Search [JavaScript]

### LINKED ISSUE #294: Linear Search [JavaScript]
#### This issue is about [Linear Search](https://en.wikipedia.org/wiki/Linear_search).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by Anshuman-Verma**
Fixes #294 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: linear_search/linear_search.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	return -1;
+}
+
+function main () {
+	A = [];
+	for (let i = 1; i<=100; i++)
+		A.push(i);
+	index = linear_search(A, 12);
+	if (index == -1) {
+		console.log("Element was not found");
```
Comment (aashutoshrathi) [inline linear_search/linear_search.js]: Use single quotes in string in whole code.

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: Alright

**Comment by Anshuman-Verma** [comment]
@aashutoshrathi looks good now ?

File: linear_search/linear_search.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	}
+	return -1;
+}
+
+function main () {
+	let A = [];
+	for (let i = 1; i<=100; i++) {
+		A.push(i);
+	}
+	index = linearSearch(A, 12);
```
Comment (aashutoshrathi) [inline linear_search/linear_search.js]: Shouldn't it be `var index = linearSearch(A, 12);`

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: damn, forgot that.
Though @singhpratyush suggested to use let

**Comment by Anshuman-Verma** [comment]
Any more flaws ? @aviaryan @aashutoshrathi

File: linear_search/linear_search.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		if (arr[it] === x) {
+			return it;
+		}
+	}
+	return -1;
+}
+
+function main () {
+	let A = [];
+	for (let i = 1; i<=100; i++) {
```
Comment (aashutoshrathi) [inline linear_search/linear_search.js]: spaces around `=<`

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: Fixed

Comment (aashutoshrathi) [inline linear_search/linear_search.js]: Still not fixed ??

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: Fixed already

Comment (aashutoshrathi) [inline linear_search/linear_search.js]: harkuchi

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: You are looking at the old commit, this is the latest https://github.com/iiitv/algos/pull/303/commits/80b47a7dcb6c77574e971d26473440fcd3f254e1

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: gitroll
look at the commit timings and my commenting time

Comment (aashutoshrathi) [inline linear_search/linear_search.js]: In line 18 
`for (let i = 1; i<=100; i++) {`
change it to 
`for (let i = 1; i <= 100; i++) {`

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: Okay my bad, thought u talkin about function.

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: Fixed now finally, everything should be okay now :laughing:

File: linear_search/linear_search.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	for (let it = 0; it < arr.length; it++) {
+		if (arr[it] === x) {
+			return it;
+		}
+	}
+	return -1;
+}
+
+function main () {
+	let A = [];
```
Comment (mohitkyadav) [inline linear_search/linear_search.js]: Don't use ALLCAPS for local variables and use a general word instead of `A`, i.e. `let data =[];`

Comment (Anshuman-Verma) [inline linear_search/linear_search.js]: FIXED

File: linear_search/linear_search.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+function linearSearch (arr, x) {
+	/*
+	Performs a linear search
+	:param arr: List of elements to search from
+	:param x: Element to search for
+	:return: Index if element found else -1
+	*/
+	for (let it = 0; it < arr.length; it++) {
+		if (arr[it] === x) {
```
Comment (aviaryan) [inline linear_search/linear_search.js]: @Anshuman-Verma Try `(x === arr[it])` .

Comment (aviaryan) [inline linear_search/linear_search.js]: If it doesn't work out, we can remove that issue.

**Comment by Anshuman-Verma** [comment]
@aviaryan I think this is it.

**Pull request merged**